src/main/java/anon/demonx/DemonX.java

```
@@ -37,164 +37,143 @@
+        jmxPasswordOpt.setRequired(false);
+        options.addOption(jmxPasswordOpt);
 
-    try {
-      cmd = parser.parse(options, args);
-    } catch (ParseException e) {
-      System.out.println(e.getMessage());
-      formatter.printHelp("demonX help", options);
+        CommandLineParser parser = new DefaultParser();
+        HelpFormatter formatter = new HelpFormatter();
+        CommandLine cmd;
 
-      System.exit(1);
-      return;
-    }
+        try {
+            cmd = parser.parse(options, args);
+        } catch (ParseException e) {
+            System.out.println(e.getMessage());
+            formatter.printHelp("demonX help", options);
 
-    String jmxHost = cmd.getOptionValue("host");
-    Integer jmxPort = Integer.valueOf(cmd.getOptionValue("port"));
-    String jmxUsername = cmd.getOptionValue("username");
-    String jmxPassword = cmd.getOptionValue("password");
+            System.exit(1);
+            return;
+        }
 
-    if ((jmxPort < 1) || (jmxPort > 65535)) {
-      System.out.println("Port must be between 1 and 65535");
-      System.exit(1);
-    }
+        String jmxHost = cmd.getOptionValue("host");
+        Integer jmxPort = Integer.valueOf(cmd.getOptionValue("port"));
+        String jmxUsername = cmd.getOptionValue("username");
+        String jmxPassword = cmd.getOptionValue("password");
 
-    // Credentials
-    HashMap   environment = new HashMap();
-    String[]  credentials = new String[] {jmxUsername, jmxPassword};
-    environment.put(JMXConnector.CREDENTIALS, credentials);
-
-    // Start connection to JMX
-    JMXServiceURL url = new JMXServiceURL(String.format("service:jmx:rmi:///jndi/rmi://%s:%d/jmxrmi", new Object[] { jmxHost, jmxPort }));
-    JMXConnector jmxc = null;
-    try {
-      jmxc = JMXConnectorFactory.connect(url, environment);
-    } catch (IOException e) {
-      System.out.println(e);
-      System.exit(1);
-    } catch (Exception e) {
```

Why did you remove the catch all exception?